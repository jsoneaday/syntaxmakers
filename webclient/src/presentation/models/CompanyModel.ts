import { Key } from "react";
import { Company } from "../../domain/repository/CompanyRepo";
/// @ts-ignore
import { v4 as uuidv4 } from "uuid";

type CompanyModel = {
  key: Key;
  id: string;
  updatedAt: string;
  name: string;
};
export default CompanyModel;

export function convert(company: Company): CompanyModel {
  return {
    key: uuidv4(),
    id: company.id,
    updatedAt: company.updatedAt,
    name: company.name,
  };
}
