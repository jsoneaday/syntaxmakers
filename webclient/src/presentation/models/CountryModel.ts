import { Key } from "react";
import { Country } from "../../domain/repository/CountryRepo";
/// @ts-ignore
import { v4 as uuidv4 } from "uuid";

type CountryModel = {
  key: Key;
  id: string;
  updatedAt: string;
  name: string;
};
export default CountryModel;

export function convert(country: Country): CountryModel {
  return {
    key: uuidv4(),
    id: country.id,
    updatedAt: country.updatedAt,
    name: country.name,
  };
}
