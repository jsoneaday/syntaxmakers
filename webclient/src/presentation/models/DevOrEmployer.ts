export enum DevOrEmployer {
  Developer = "Developer",
  Employer = "Employer",
}
