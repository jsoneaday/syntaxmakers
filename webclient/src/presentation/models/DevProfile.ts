import { Key } from "react";
import { Developer } from "../../domain/repository/DeveloperRepo";
/// @ts-ignore
import { v4 as uuidv4 } from "uuid";

type DevProfile = {
  key: Key;
  id: string; // bigint is not serializable by Redux
  updatedAt: string;
  userName: string;
  fullName: string;
  email: string;
  primaryLangId: number;
  secondaryLangId?: number | null;
  accessToken?: string | null;
};
export default DevProfile;

export function convert(dev: Developer, accessToken?: string): DevProfile {
  return {
    key: uuidv4(),
    id: dev.id, // bigint is not serializable by Redux
    updatedAt: dev.updatedAt,
    userName: dev.userName,
    fullName: dev.fullName,
    email: dev.email,
    primaryLangId: dev.primaryLangId,
    secondaryLangId: dev.secondaryLangId,
    accessToken,
  };
}
