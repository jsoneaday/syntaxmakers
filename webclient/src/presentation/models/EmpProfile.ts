import { Key } from "react";
/// @ts-ignore
import { v4 as uuidv4 } from "uuid";
import { Employer } from "../../domain/repository/EmployerRepo";

type EmpProfile = {
  key: Key;
  id: string;
  updatedAt: string;
  userName: string;
  fullName: string;
  email: string;
  companyId: string;
  accessToken?: string | null;
};
export default EmpProfile;

export function convert(emp: Employer, accessToken?: string): EmpProfile {
  return {
    key: uuidv4(),
    id: emp.id,
    updatedAt: emp.updatedAt,
    userName: emp.userName,
    fullName: emp.fullName,
    email: emp.email,
    companyId: emp.companyId,
    accessToken,
  };
}
