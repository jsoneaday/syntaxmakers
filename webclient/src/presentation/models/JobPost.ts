import { Key } from "react";
import { KeyItem } from "../common/utils";

export default class JobPost implements KeyItem {
  constructor(
    public key: Key,
    public id: bigint,
    public updatedAt: string,
    public title: string,
    public description: string,
    public companyName: string,
    public isRemote: boolean,
    public primaryLangName: string,
    public secondaryLangName: string,
    public industryName: String,
    public salary: string,
    public logo?: Blob,
    public countryName?: string
  ) {}
}
