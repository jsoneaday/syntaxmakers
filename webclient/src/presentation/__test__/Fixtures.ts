import DevProfile from "../models/DevProfile";
/// @ts-ignore
import { v4 as uuidv4 } from "uuid";
import { faker } from "@faker-js/faker";
import { getFakeFullName } from "../../domain/TestUtils";

export const defaultDevProfile: DevProfile = {
  key: uuidv4(),
  id: "1",
  updatedAt: new Date().toISOString(),
  userName: "ronaldo", //faker.internet.userName(),
  fullName: getFakeFullName(),
  email: faker.internet.email(),
  primaryLangId: "1",
};
