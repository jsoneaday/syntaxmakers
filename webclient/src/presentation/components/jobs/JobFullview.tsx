import { useLocation } from "react-router-dom";
import JobPost from "../../models/JobPost";
import { ChangeEvent, useEffect, useState } from "react";
import "../../theme/job_full_view.css";
import { appendPlusLargeCurrency } from "../../common/CurrencyFormatter";
import flag from "../../theme/assets/flag.png";
import similar from "../../theme/assets/similar.png";
import GoBack from "../../components/navigation/GoBack";
import TextEditor from "../../components/textEditor/TextEditor";
import DropDown from "../controls/DropDown";

interface JobFullviewProps {
  readOnly: boolean;
}

export default function JobFullview({ readOnly }: JobFullviewProps) {
  const { state } = useLocation();
  const [jobPost, setJobPost] = useState<JobPost>();
  const [salary, setSalary] = useState("");
  const [title, setTitle] = useState("");
  const [companyName, setCompanyName] = useState("");
  const [isRemote, setIsRemote] = useState(false);
  const [updatedAt, setUpdatedAt] = useState("");

  useEffect(() => {
    const currentJobPost = state as JobPost;
    setJobPost(currentJobPost);
    setSalary(appendPlusLargeCurrency(currentJobPost?.salary || ""));
  }, [state]);

  const onChangeTitle = (e: ChangeEvent<HTMLInputElement>) => {
    e.preventDefault();
  };

  return (
    <form className="userhome-main" style={{ margin: "auto" }}>
      <div
        style={{ paddingTop: "2em", paddingLeft: "2em", paddingRight: "2em" }}
      >
        <GoBack label="dev home" />
      </div>
      <div
        className="opposites"
        style={{ paddingTop: "2em", paddingLeft: "2em", paddingRight: "2em" }}
      >
        <div className="userhome-top">
          <div className="opposites">
            {readOnly ? (
              <div className="title-font">{jobPost?.title}</div>
            ) : (
              <div className="left-align">
                <label htmlFor="job-title-input" style={{ marginRight: "1em" }}>
                  Title
                </label>
                <input
                  id="job-title-input"
                  type="text"
                  value={title}
                  onChange={onChangeTitle}
                  className="input"
                />
              </div>
            )}
          </div>

          <div className="opposites">
            <div className="job-full-view-subtitle">
              {readOnly ? (
                <div className="sub-title-font job-full-view-subtitle-item-primary">
                  {jobPost?.companyName}
                </div>
              ) : (
                <DropDown optionItems={[{ name: "Company A", value: "1" }]} />
              )}
              <div className="sub-title-font job-full-view-subtitle-item-primary">
                {jobPost?.isRemote ? "Remote" : jobPost?.countryName}
              </div>
              <div className="sub-title-font job-full-view-subtitle-item-primary">
                {jobPost?.updatedAt}
              </div>
            </div>
          </div>

          <div className="job-full-view-subtitle-item-secondary">
            {`Contact: ${jobPost?.employerName}`}
          </div>
          <div className="job-full-view-subtitle-item-secondary">
            {`Primary Language: ${jobPost?.primaryLangName}`}
          </div>
          {jobPost?.secondaryLangName &&
          jobPost?.secondaryLangName != jobPost?.primaryLangName ? (
            <div className="job-full-view-subtitle-item-secondary">
              {`Secondary Language: ${jobPost?.secondaryLangName}`}
            </div>
          ) : null}
          <div className="job-full-view-subtitle-item-secondary">
            {`Industry: ${jobPost?.industryName}`}
          </div>
          <div className="job-full-view-subtitle-item-secondary">
            {`Base Salary: ${salary}`}
          </div>
        </div>
        <div
          className="stack"
          style={{ alignItems: "flex-end", textAlign: "right" }}
        >
          <button
            className="primary-btn small-btn"
            style={{ marginBottom: ".5em" }}
          >
            apply
          </button>
          <button className="secondary-btn small-btn">save</button>
          <img
            src={flag}
            className="job-icon"
            style={{ marginTop: "1em" }}
            title="inappropriate"
          />
          <img
            src={similar}
            className="job-icon"
            style={{ marginTop: ".50em" }}
            title="similar jobs"
          />
        </div>
      </div>

      <div
        className="normal-font dev-post-preview-container"
        style={{
          padding: "2em",
        }}
      >
        <span className="title-font" style={{ marginBottom: "1em" }}>
          Description
        </span>
        <TextEditor
          initialValue={[
            {
              type: "paragraph",
              children: [{ text: "A line of text in a paragraph." }],
            },
          ]}
          readOnly={readOnly}
        />
      </div>
    </form>
  );
}
