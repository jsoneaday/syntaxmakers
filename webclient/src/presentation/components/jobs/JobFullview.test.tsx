import JobPost from "../../models/JobPost";
/// @ts-ignore
import { v4 as uuidv4 } from "uuid";
import { render, screen } from "@testing-library/react";
import JobFullview from "./JobFullview";
import { RouterProvider, createBrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { defaultDevProfile } from "../../__test__/Fixtures";
import { DevOrEmployer } from "../../models/DevOrEmployer";

const router = createBrowserRouter([
  {
    path: "/",
    element: <JobFullview readOnly={true} />,
  },
]);

const title = "Senior JavaScript Developer";
const employerName = "John Dole";
const isRemote = true;
const companyName = "Acme Inc";
const primaryLangName = "JavaScript";
const secondaryLangName = "C#";
const industryName = "Finance";
const salary = "200000";

const jobPost: JobPost = {
  key: uuidv4(),
  id: "1",
  updatedAt: "2022/01/03T22:21:02",
  title,
  description: "A description",
  employerId: "1",
  employerName,
  companyId: "1",
  companyName,
  isRemote,
  primaryLangId: "1",
  primaryLangName,
  secondaryLangId: "1",
  secondaryLangName,
  industryId: "1",
  industryName,
  salaryId: "1",
  salary,
};

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => {
    return {
      state: jobPost,
      key: "123",
      pathname: "",
      search: "",
      hash: "",
    };
  },
}));

describe("Test JobFullview", () => {
  it("matches snapshot", async () => {
    const mockStore = configureStore();
    const store = mockStore({
      profile: defaultDevProfile,
      devOrEmployer: DevOrEmployer.Developer,
    });

    const result = render(
      <Provider store={store}>
        <RouterProvider router={router} />
      </Provider>
    );

    expect(result).toMatchSnapshot();
  });

  it("has correct title", async () => {
    const mockStore = configureStore();
    const store = mockStore({
      profile: defaultDevProfile,
      devOrEmployer: DevOrEmployer.Developer,
    });

    render(
      <Provider store={store}>
        <RouterProvider router={router} />
      </Provider>
    );

    screen.getByText(title);
    screen.getByText(`Contact ${employerName}`);
    screen.getByText("Remote");
    screen.getByText(companyName);
    screen.getByText(`Primary Language ${primaryLangName}`);
    screen.getByText(`Secondary Language ${secondaryLangName}`);
    screen.getByText(`Industry ${industryName}`);
    screen.getByText(`Base Salary ${salary}`);
  });
});
