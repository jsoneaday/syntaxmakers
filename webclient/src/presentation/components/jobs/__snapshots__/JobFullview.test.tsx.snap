// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test JobFullview matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="userhome-main"
        style="margin: auto;"
      >
        <div
          class="header-container job-full-view-header"
          style="padding-top: 2em; padding-left: 2em; padding-right: 2em;"
        >
          <div
            class="goback-container"
          >
            <img
              class="goback-img"
              src="default"
              title="Go back"
            />
            <label>
              developer home
            </label>
          </div>
        </div>
        <div
          class="opposites"
          style="padding-top: 2em; padding-left: 2em; padding-right: 2em;"
        >
          <div
            class="stack"
          >
            <div
              class="title-font"
            >
              Senior JavaScript Developer
            </div>
            <div
              class="left-align"
            >
              <div
                class="opposites"
              >
                <div
                  class="job-full-view-subtitle"
                  style="width: 100%;"
                >
                  <div
                    style="margin-bottom: .5em;"
                  >
                    <div
                      class="sub-title-font job-full-view-subtitle-item-primary"
                    >
                      Acme Inc
                    </div>
                  </div>
                  <div
                    class="sub-title-font job-full-view-subtitle-item-primary"
                  >
                    Remote
                  </div>
                  <div
                    class="small-font job-full-view-subtitle-item-primary"
                  >
                    2022/01/03T22:21:02
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="stack"
            style="align-items: flex-end; text-align: right;"
          >
            <button
              class="primary-btn small-btn"
              style="margin-bottom: .5em;"
            >
              apply
            </button>
            <button
              class="secondary-btn small-btn"
            >
              save
            </button>
            <img
              class="job-icon"
              src="default"
              style="margin-top: 1em;"
              title="inappropriate"
            />
            <img
              class="job-icon"
              src="default"
              style="margin-top: .50em;"
              title="similar jobs"
            />
          </div>
        </div>
        <div
          class="job-full-view-section"
          style="padding: 1.5em; margin-bottom: 1em;"
        >
          <div
            class="job-full-view-subtitle-item-secondary"
          >
            Contact John Dole
          </div>
          <div
            class="job-full-view-subtitle-item-secondary"
          >
            Primary Language JavaScript
          </div>
          <div
            class="job-full-view-subtitle-item-secondary"
          >
            Secondary Language C#
          </div>
          <div
            class="job-full-view-subtitle-item-secondary"
          >
            Industry Finance
          </div>
          <div
            class="job-full-view-subtitle-item-secondary"
          >
            Base Salary 200000
          </div>
        </div>
        <div
          class="normal-font dev-post-preview-container job-full-view-section"
          style="padding: 2em;"
        >
          <span
            class="title-font"
            style="margin-bottom: 1em;"
          >
            Description
          </span>
          <div
            autocapitalize="false"
            autocorrect="false"
            class="txtedit-container"
            contenteditable="false"
            data-slate-editor="true"
            data-slate-node="value"
            spellcheck="false"
            style="position: relative; white-space: pre-wrap; word-wrap: break-word;"
            zindex="-1"
          >
            <p
              attributes="[object Object]"
              element="[object Object]"
            >
              <span
                data-slate-node="text"
              >
                <span
                  data-slate-leaf="true"
                  style="font-weight: normal;"
                >
                  <span
                    data-slate-string="true"
                  >
                    Testing Text
                  </span>
                </span>
              </span>
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="userhome-main"
      style="margin: auto;"
    >
      <div
        class="header-container job-full-view-header"
        style="padding-top: 2em; padding-left: 2em; padding-right: 2em;"
      >
        <div
          class="goback-container"
        >
          <img
            class="goback-img"
            src="default"
            title="Go back"
          />
          <label>
            developer home
          </label>
        </div>
      </div>
      <div
        class="opposites"
        style="padding-top: 2em; padding-left: 2em; padding-right: 2em;"
      >
        <div
          class="stack"
        >
          <div
            class="title-font"
          >
            Senior JavaScript Developer
          </div>
          <div
            class="left-align"
          >
            <div
              class="opposites"
            >
              <div
                class="job-full-view-subtitle"
                style="width: 100%;"
              >
                <div
                  style="margin-bottom: .5em;"
                >
                  <div
                    class="sub-title-font job-full-view-subtitle-item-primary"
                  >
                    Acme Inc
                  </div>
                </div>
                <div
                  class="sub-title-font job-full-view-subtitle-item-primary"
                >
                  Remote
                </div>
                <div
                  class="small-font job-full-view-subtitle-item-primary"
                >
                  2022/01/03T22:21:02
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="stack"
          style="align-items: flex-end; text-align: right;"
        >
          <button
            class="primary-btn small-btn"
            style="margin-bottom: .5em;"
          >
            apply
          </button>
          <button
            class="secondary-btn small-btn"
          >
            save
          </button>
          <img
            class="job-icon"
            src="default"
            style="margin-top: 1em;"
            title="inappropriate"
          />
          <img
            class="job-icon"
            src="default"
            style="margin-top: .50em;"
            title="similar jobs"
          />
        </div>
      </div>
      <div
        class="job-full-view-section"
        style="padding: 1.5em; margin-bottom: 1em;"
      >
        <div
          class="job-full-view-subtitle-item-secondary"
        >
          Contact John Dole
        </div>
        <div
          class="job-full-view-subtitle-item-secondary"
        >
          Primary Language JavaScript
        </div>
        <div
          class="job-full-view-subtitle-item-secondary"
        >
          Secondary Language C#
        </div>
        <div
          class="job-full-view-subtitle-item-secondary"
        >
          Industry Finance
        </div>
        <div
          class="job-full-view-subtitle-item-secondary"
        >
          Base Salary 200000
        </div>
      </div>
      <div
        class="normal-font dev-post-preview-container job-full-view-section"
        style="padding: 2em;"
      >
        <span
          class="title-font"
          style="margin-bottom: 1em;"
        >
          Description
        </span>
        <div
          autocapitalize="false"
          autocorrect="false"
          class="txtedit-container"
          contenteditable="false"
          data-slate-editor="true"
          data-slate-node="value"
          spellcheck="false"
          style="position: relative; white-space: pre-wrap; word-wrap: break-word;"
          zindex="-1"
        >
          <p
            attributes="[object Object]"
            element="[object Object]"
          >
            <span
              data-slate-node="text"
            >
              <span
                data-slate-leaf="true"
                style="font-weight: normal;"
              >
                <span
                  data-slate-string="true"
                >
                  Testing Text
                </span>
              </span>
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
